### PR TITLE
Nicer errors for including/not including ActiveRecordTransition

### LINKED ISSUE #164: Bogus IncompatibleSerializationError with PostgreSQL json on Rails 4.2.1
Using Rails 4.2.1, pg 0.18.1 and statesman 1.2.3 against PostgreSQL 9.4.2, I receive a `Statesman::IncompatibleSerializationError` when trying to store the transitions' `metadata` in a `json` column (the same error occurs trying to store in a `jsonb` column. I have also tested this against rails/rails 4-2-stable branch.

```
irb(main):001:0> Dummy.new.state_machine.transition_to(:intermediate)
Statesman::IncompatibleSerializationError: DummyTransition#metadata column type cannot be json
                  and serialized simultaneously
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/statesman-1.2.3/lib/statesman/adapters/active_record.rb:27:in `initialize'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/statesman-1.2.3/lib/statesman/machine.rb:185:in `new'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/statesman-1.2.3/lib/statesman/machine.rb:185:in `initialize'
        from /tmp/statesman-serializationerror/app/models/dummy.rb:11:in `new'
        from /tmp/statesman-serializationerror/app/models/dummy.rb:11:in `state_machine'
        from (irb):1
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/console.rb:110:in `start'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/console.rb:9:in `start'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:68:in `console'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.1/lib/rails/commands.rb:17:in `<top (required)>'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
        from /tmp/statesman-serializationerror/bin/rails:8:in `<top (required)>'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:268:in `load'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:268:in `block in load'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:268:in `load'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/niels/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from -e:1:in `<main>'
irb(main):003:0* DummyTransition.columns_hash["metadata"]
=> #<ActiveRecord::ConnectionAdapters::PostgreSQLColumn:0x007f04b85bec58 @array=false, @name="metadata", @cast_type=#<ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Json:0x007f04b8a10f40 @precision=nil, @scale=nil, @limit=nil>, @sql_type="json", @null=true, @default=nil, @default_function=nil>
irb(main):004:0> DummyTransition.columns_hash["metadata"].cast_type
=> #<ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Json:0x007f04b8a10f40 @precision=nil, @scale=nil, @limit=nil>
irb(main):005:0> DummyTransition.columns_hash["metadata"].cast_type.is_a?(::ActiveRecord::Type::Serialized)
=> true
irb(main):006:0> DummyTransition.columns_hash["metadata"].cast_type.class.is_a?(::ActiveRecord::Type::Serialized)
=> false
```

# Demo app

See https://github.com/herimedia/statesman-serializationerror for a reduced app showing the problem or below for the non-vanilla files:

## db/schema.rb

```ruby
ActiveRecord::Schema.define(version: 20150615172835) do

  # These are extensions that must be enabled in order to support this database
  enable_extension "plpgsql"

  create_table "dummies", force: :cascade do |t|
  end

  create_table "dummy_transitions", force: :cascade do |t|
    t.string   "to_state",    null: false
    t.json     "metadata"
    t.integer  "sort_key",    null: false
    t.integer  "dummy_id",    null: false
    t.boolean  "most_recent", null: false
    t.datetime "created_at",  null: false
    t.datetime "updated_at",  null: false
  end

  add_index "dummy_transitions", ["dummy_id", "most_recent"], name: "index_dummy_transitions_parent_most_recent", unique: true, where: "most_recent", using: :btree
  add_index "dummy_transitions", ["dummy_id", "sort_key"], name: "index_dummy_transitions_parent_sort", unique: true, using: :btree

end
```

## app/models/dummy.rb

```ruby
class Dummy < ActiveRecord::Base

  include Statesman::Adapters::ActiveRecordQueries

  has_many :dummy_transitions,
    class_name:   "DummyTransition",
    dependent:    :destroy,
    inverse_of:   :dummy

  def state_machine
    @state_machine ||= DummyStateMachine.new(self, transition_class: DummyTransition)
  end

  class << self

    private

    def initial_state
      :new
    end

    def transition_class
      DummyTransition
    end

  end

end
```

## app/models/dummy_transition.rb

```ruby
class DummyTransition < ActiveRecord::Base

  include Statesman::Adapters::ActiveRecordTransition

  belongs_to :dummy,
    class_name: "Dummy",
    inverse_of: :dummy_transitions

end
```

## app/models/dummy_state_machine.rb

```ruby
class DummyStateMachine

  include Statesman::Machine

  state :new, initial: true
  state :intermediate
  state :final

  transition from: :new,          to: [:intermediate, :final]
  transition from: :intermediate, to: [:final]

end
```
## config/initializers/statesman.rb

```ruby
Statesman.configure do
  storage_adapter(Statesman::Adapters::ActiveRecord)
end
```

### PR DESCRIPTION
**Pull request by isaacseymour**
The solution is always to start/stop `include Statesman::Adapters::ActiveRecordTransition`-ing. So we should probably mention that. Resolves #164

### CODE REVIEW

**Comment by isaacseymour** [comment]
@petehamilton could you review?

File: .rubocop.yml [hunk 1]
```diff
@@ -42,3 +42,7 @@ SingleSpaceBeforeFirstArg:
 
 DotPosition:
   EnforcedStyle: trailing
+
+# Allow class and message or instance raises
+Style/RaiseArgs:
```
Comment (petehamilton) [inline .rubocop.yml]: Why is this necessary?

Comment (isaacseymour) [inline .rubocop.yml]: See the changes - the error messages have now got too long for rubocop to be happy with having both of them in a message, so I've moved the main bits of the message out into the error classes themselves. I think it's nicer this way (the errors are single-use only anyway)

Comment (petehamilton) [inline .rubocop.yml]: I'm possibly misunderstanding, but wouldn't `raise IncompatibleSerializationError.new, transition_class.name` be fine?

Comment (petehamilton) [inline .rubocop.yml]: I was misunderstanding. I'm also blind. :+1:

**Comment by petehamilton** [comment]
:+1:
